r2/r2/lib/media.py

```diff
@@ -312,6 +312,16 @@ def _scrape_media(url, autoplay=False, maxwidth=600, force=False,
             print "%s made a bad secure media obj for url %s" % (scraper, url)
             secure_media_object = None
 
+        # If thumbnail can't be found, attempt again using _ThumbnailOnlyScraper
+        # This should fix bugs that occur when embed.ly caches links before the thumbnail is available
+        if not thumbnail_image:
+            scraper = _ThumbnailOnlyScraper(url)
```

Comment:
This will also recreate the thumbnail scraper if that's what we were already using in the first place, no?

My initial thought on how we should handle this situation is to make [`for_url()`](https://github.com/reddit/reddit/blob/master/r2/r2/lib/media.py#L492-L504) be a generator instead of returning a singular value - so that if one scraper fails, we can fall through to the next appropriate one, whatever that may be, and keep doing that until we succeed at finding something or there aren't any more scrapers.